packages/fx-core/src/plugins/resource/bot/configs/scaffoldConfig.ts

```diff
@@ -2,55 +2,48 @@
 // Licensed under the MIT license.
 import * as utils from "../utils/common";
 import { CommonStrings, PluginBot } from "../resources/strings";
+import { isMultiEnvEnabled } from "../../../../common";
 import { PluginContext } from "@microsoft/teamsfx-api";
 import { ProgrammingLanguage } from "../enums/programmingLanguage";
-import { WayToRegisterBot } from "../enums/wayToRegisterBot";
 import path from "path";
 
 export class ScaffoldConfig {
   public botId?: string;
   public botPassword?: string;
   public objectId?: string;
   public programmingLanguage?: ProgrammingLanguage;
-  public wayToRegisterBot?: WayToRegisterBot;
   public workingDir?: string;
 
-  public botRegistrationCreated(): boolean {
-    if (this.botId && this.botPassword && this.objectId) {
+  public botAADCreated(): boolean {
+    if (this.botId && this.botPassword) {
       return true;
     }
-
+    // todo: valid this appid and password
     return false;
   }
 
   public async restoreConfigFromContext(context: PluginContext): Promise<void> {
     this.workingDir = path.join(context.root, CommonStrings.BOT_WORKING_DIR_NAME);
-
     this.botId = context.config.get(PluginBot.BOT_ID) as string;
-
     this.botPassword = context.config.get(PluginBot.BOT_PASSWORD) as string;
-
     this.objectId = context.config.get(PluginBot.OBJECT_ID) as string;
+    if (isMultiEnvEnabled()) {
+      this.botId = context.envInfo.config.bot?.appId ?? this.botId;
+      this.botPassword = context.envInfo.config.bot?.appPassword ?? this.botPassword;
+    }
 
     const rawProgrammingLanguage = context.projectSettings?.programmingLanguage;
-
     if (
       rawProgrammingLanguage &&
       utils.existsInEnumValues(rawProgrammingLanguage, ProgrammingLanguage)
     ) {
       this.programmingLanguage = rawProgrammingLanguage as ProgrammingLanguage;
     }
-
-    const rawWay = context.config.get(PluginBot.WAY_TO_REGISTER_BOT) as string;
-    if (rawWay && utils.existsInEnumValues(rawWay, WayToRegisterBot)) {
-      this.wayToRegisterBot = rawWay as WayToRegisterBot;
-    }
   }
 
   public saveConfigIntoContext(context: PluginContext): void {
```

Comment:
Since input is from config file, do we still need this method?
For deployment step, it need to consume output of provision to get this information?